#!/bin/bash
#PBS -N chr22_filt
#PBS -j oe
#PBS -l select=1:ncpus=3:mem=32gb -l place=free
#PBS -m abe
#PBS -M phillevn@gmail.com


cd ${PBS_O_WORKDIR}

module load gcc/9.2.0

# /ddn/home2/r2725/apps/htslib-1.12/bin/tabix -p vcf /ddn/home2/r2725/projects/jeannette/data/chr1.dose.vcf.gz

for i in {1..20}; do

	cd /ddn/home2/r2725/projects/jeannette/data/

	mkdir chr${i}_data

	cd chr${i}_data

	mkdir subset_chro${i}_AA_filtered

	mkdir subset_chro${i}_EA_filtered


	# subset the chromosome with the same topid from selected phenotype
	bcftools view -Oz -S /ddn/home2/r2725/projects/jeannette/data/AA_gwas_id.txt /ds/home/r1952/TOPMED_GWAS/AA/chr${i}.dose.vcf.gz -o /ddn/home2/r2725/projects/jeannette/data/chr${i}_data/subset_chro${i}_AA.vcf.gz
	# for EA 
	bcftools view -Oz -S /ddn/home2/r2725/projects/jeannette/data/EA_gwas_id.txt /ds/home/r1952/TOPMED_GWAS/EA/chr${i}.dose.vcf.gz -o /ddn/home2/r2725/projects/jeannette/data/chr${i}_data/subset_chro${i}_EA.vcf.gz

	# run Plink2 to converse vcf to filter maf >0.01 and < 0.999 file 

	/ddn/home2/r2725/apps/plink/plink2.0/plink2 --vcf /ddn/home2/r2725/projects/jeannette/data/chr${i}_data/subset_chro${i}_AA.vcf.gz --maf 0.01 --max-maf 0.99 --max-alleles 2 --min-alleles 2 --id-delim  --recode vcf --out /ddn/home2/r2725/projects/jeannette/data/chr${i}_data/subset_chro${i}_AA_filtered

	/ddn/home2/r2725/apps/plink/plink2.0/plink2 --vcf /ddn/home2/r2725/projects/jeannette/data/chr${i}_data/subset_chro${i}_EA.vcf.gz --maf 0.01 --max-maf 0.99 --max-alleles 2 --min-alleles 2 --id-delim  --recode vcf --out /ddn/home2/r2725/projects/jeannette/data/chr${i}_data/subset_chro${i}_EA_filtered


	# Split the chromosome into smaller files with 1,000,000 snps and below
	#grab the header
	head -n 100 subset_chro${i}_AA_filtered.vcf | grep "^#" >header
	#grab the non header lines
	grep -v "^#" subset_chro${i}_AA_filtered.vcf >variants
	#split into chunks with 1000 lines
	split -l 1000000 variants subset_chro${i}_AA_filtered_split
	#reattach the header to each and clean up

	for j in subset_chro${i}_AA_filtered_split*;
		do cat header $j >$j.vcf && gzip -k $j.vcf  && rm -f $j &&
			/ddn/home2/r2725/apps/plink/plink2.0/plink2 --vcf ${j}.vcf.gz --id-delim --make-bed --out /ddn/home2/r2725/projects/jeannette/data/chr${i}_data/${j}_bed;
		done
	rm -f header variants

#	mkdir subset_chro${i}_AA_filtered_split # dont't need

	mv subset_chro${i}_AA_filtered_split* ./subset_chro${i}_AA_filtered/

	# going to EA diretory

#	cd ../subset_chro${i}_EA_filtered

	# Split the chromosome into smaller files with 1,000,000 snps and below
	#grab the header
	head -n 100 subset_chro${i}_EA_filtered.vcf | grep "^#" >header
	#grab the non header lines
	grep -v "^#" subset_chro${i}_EA_filtered.vcf >variants
	#split into chunks with 1000 lines
	split -l 1000000 variants subset_chro${i}_EA_filtered_split
	#reattach the header to each and clean up
	for j in subset_chro${i}_EA_filtered_split*;
		do cat header $j >$j.vcf && gzip -k $j.vcf &&  rm -f $j &&
			/ddn/home2/r2725/apps/plink/plink2.0/plink2 --vcf ${j}.vcf.gz --id-delim --make-bed --out /ddn/home2/r2725/projects/jeannette/data/chr${i}_data/${j}_bed;
		done
	rm -f header variants


	mv subset_chro${i}_EA_filtered_split* ./subset_chro${i}_EA_filtered/;
done

